File: cpp/452_用最少数量的箭引爆气球_medium.cpp
class Solution {
public:
    int findMinArrowShots(vector<vector<int>>& points) {
        if(points.empty())
            return 0;
        sort(points.begin(), points.end(), [](const vector<int>& u, const vector<int>& v){
            return u[1]<v[1];
        });
        int ans = 1;
        int arrow_pos = points[0][1];
        for(const vector<int> &ballow:points)
            if(ballow[0] > arrow_pos){
                arrow_pos = ballow[1];
                ans++;
            }
        return ans;
    }
};

File: python/452_用最少数量的箭引爆气球_medium.py
class Solution:
    def findMinArrowShots(self, points: List[List[int]]) -> int:
        if not points:
            return 0
        points.sort(key=lambda points:points[1])
        ans = 1
        arrow = points[0][1]
        for ballon in points:
            if ballon[0] > arrow:
                arrow = ballon[1]
                ans += 1
        return ans